vscode-dotnet-runtime-library/src/Acquisition/IDistroDotnetSDKProvider.ts

```diff
@@ -0,0 +1,173 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import * as fs from 'fs';
+import { DistroVersionPair, DotnetDistroSupportStatus } from './LinuxVersionResolver';
+import path = require('path');
+import { DotnetAcquisitionDistroUnknownError } from '../EventStream/EventStreamEvents';
+import { VersionResolver } from './VersionResolver';
+import { ICommandExecutor } from '../Utils/ICommandExecutor';
+import { CommandExecutor } from '../Utils/CommandExecutor';
+/* tslint:disable:no-any */
+
+/**
+ * This interface describes the functionality needed to manage the .NET SDK on a specific distro and version of Linux.
+ *
+ * @remarks We accept community contributions of this interface for each distro-version pair.
+ * All calls which require sudo must leverage the vscode/sudo library. We will not accept contributions that use other methods to gain admin privellege.
+ * Please see DotnetDistroVersion as well to add your version.
+ */
+export abstract class IDistroDotnetSDKProvider {
+
+    protected commandRunner : ICommandExecutor;
+    protected distroVersion : DistroVersionPair;
+    protected distroJson : any | null = null;
+
+    protected preinstallCommandKey = 'preInstallCommands';
+    protected installCommandKey = 'installCommand';
+    protected uninstallCommandKey = 'uninstallCommand';
+    protected updateCommandKey = 'updateCommand';
+    protected packageLookupCommandKey = 'packageLookupCommand';
+    protected currentInstallPathCommandKey = 'currentInstallPathCommand';
+    protected isInstalledCommandKey = 'isInstalledCommand';
+    protected expectedMicrosoftFeedInstallDirKey = 'expectedMicrosoftFeedInstallDirectory';
+    protected expectedDistroFeedInstallDirKey = 'expectedDistroFeedInstallDirectory';
+    protected installedSDKVersionsCommandKey = 'installedSDKVersionsCommand';
+    protected installedRuntimeVersionsCommandKey = 'installedRuntimeVersionsCommand';
+    protected currentInstallVersionCommandKey = 'currentInstallationVersionCommand';
+
+    protected distroVersionsKey = 'versions';
+    protected versionKey = 'version';
+    protected dotnetPackagesKey = 'dotnet';
+    protected sdkKey = 'sdk';
+    protected runtimeKey = 'runtime';
+    protected aspNetKey = 'aspnetcore';
+
+    constructor(distroVersion : DistroVersionPair, executor : ICommandExecutor | null = null)
+    {
+        this.commandRunner = executor ?? new CommandExecutor();
+        this.distroVersion = distroVersion;
+        // Hard-code to the upper path (lib/dist/acquisition) from __dirname to the lib folder, as webpack-copy doesn't seem to copy the distro-support.json
```

Comment:
I would double check that this works with all building methods (when you run from vscode debugger, when you build a .vsix, etc), if you haven't already, since webpack can be tricky with importing content files. It's probably possible to use a file copy webpack extension to do this with webpack and import the json file directly, but if this is working for you then that's great too.

Reply:
I was able to confirm it works with the debugger and packing the extension! That was a good call though. It helped me find another related bug :)

Reply:
Ah jk, the other bug hid the fact that this wasnt working